changes/changes.go

```diff
@@ -0,0 +1,612 @@
+package changes
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	golog "github.com/ipfs/go-log"
+	"github.com/qri-io/dataset"
+	"github.com/qri-io/dataset/tabular"
+	"github.com/qri-io/qri/dsref"
+	qerr "github.com/qri-io/qri/errors"
+	"github.com/qri-io/qri/fsi"
+	"github.com/qri-io/qri/stats"
+)
+
+var (
+	log = golog.Logger("changes")
+	// ErrCompareStats indicates an error with stats comparison
+	ErrCompareStats = errors.New("stats compare error")
+)
+
+// ChangeReportComponent is a generic component used to populate the change report
+type ChangeReportComponent struct {
+	Left  interface{}            `json:"left"`
+	Right interface{}            `json:"right"`
+	Meta  map[string]interface{} `json:"meta,omitempty"`
```

Comment:
Correct me if I'm wrong, but is this only ever used to hold a `"status"` key to value? Since `"status"` appears so often (always?), maybe it's better to make that top level on this struct.

Also, I would prefer not to have this field named `Meta` since there's a dataset component with that same name, but it means something totally different. Not sure how this is used aside from the already mentioned `"status"` value, but maybe `About` would be better if it's talking about how the component has changed.

Reply:
Agree on the rename. Also for the first pass, it's true that it only holds the `"status"` but the intention is to use it for more contextual data once we get there. In that vein, I'll keep the object structure but will rename. (Depending on timelines might separate that out in another PR for timing with FE integrations)